golem-worker-service-base/src/gateway_middleware/middleware_in.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::gateway_execution::gateway_http_input_executor::GatewayHttpInput;
+use crate::gateway_execution::gateway_session::SessionId;
+use crate::gateway_middleware::HttpAuthenticationMiddleware;
+use async_trait::async_trait;
+use golem_common::SafeDisplay;
+
+#[async_trait]
+pub trait HttpMiddlewareIn<Namespace> {
```

Comment:
can we replace this with a normal function? Doesn't look like this needs to be extensible

Reply:
We have a lot of middlewares. and  the `in` and `out` middlewares are clearly distinguishable with this approach. And the next developers don't need to think where to place how to process a middleware type. It's grouped into 1 single place